router-tests/authentication_test.go

```diff
@@ -753,6 +753,175 @@ func TestAuthorization(t *testing.T) {
 			require.Equal(t, employeesExpectedData, string(data))
 		})
 	})
+
+	t.Run("token is valid if at any one JWK config validation succeeds", func(t *testing.T) {
+		t.Parallel()
+
+		authServer1, err := jwks.NewServer(t)
+		t.Cleanup(authServer1.Close)
+		require.NoError(t, err)
+
+		authServer2, err := jwks.NewServer(t)
+		t.Cleanup(authServer2.Close)
+		require.NoError(t, err)
+
+		token, err := authServer2.Token(nil)
+		require.NoError(t, err)
+
+		authenticators := ConfigureAuthWithJwksConfig(t, []authentication.JWKSConfig{
+			{
+				Secret:    "example secret",
+				Algorithm: string(jwkset.AlgHS256),
+				KeyId:     "givenKID",
+			},
+			{
+				URL:             authServer1.JWKSURL(),
+				RefreshInterval: time.Second * 5,
+			},
+			{
+				URL:             authServer2.JWKSURL(),
+				RefreshInterval: time.Second * 5,
+			},
+		})
+
+		testenv.Run(t, &testenv.Config{
+			RouterOptions: []core.Option{
+				core.WithAccessController(core.NewAccessController(authenticators, true)),
+			},
+		}, func(t *testing.T, xEnv *testenv.Environment) {
+			// Operations with a token should succeed
+			header := http.Header{
+				"Authorization": []string{"Bearer " + token},
+			}
+			res, err := xEnv.MakeRequest(http.MethodPost, "/graphql", header, strings.NewReader(employeesQuery))
+			require.NoError(t, err)
+			defer res.Body.Close()
+			require.Equal(t, http.StatusOK, res.StatusCode)
+			require.Equal(t, JwksName, res.Header.Get(xAuthenticatedByHeader))
+			data, err := io.ReadAll(res.Body)
+			require.NoError(t, err)
+			require.Equal(t, employeesExpectedData, string(data))
+		})
+	})
+
+}
+
+func TestNonHttpAuthorization(t *testing.T) {
+	t.Run("valid HS256 token", func(t *testing.T) {
```

Comment:
Can we get more descriptive texts for tests? something like `should succeed with valid token` etc to make sure you can read the test name and directly understand what your expectation is instead of reading the test code?

Reply:
Done